probe/docker/registry_test.go

```diff
@@ -183,6 +187,13 @@ func TestRegistryEvents(t *testing.T) {
 		defer registry.Stop()
 		runtime.Gosched()
 
+		check := func(want []docker.Container) {
+			test.Poll(t, 10*time.Millisecond, func() bool {
+				have := allContainers(registry)
+				return reflect.DeepEqual(want, have)
+			}, "Missed event")
```

Comment:
Yes.  We could change poll to take a function that returns a interface, and then add the test.Diff code. 

In fact, I'll do that.